app/src/main/java/com/palone/planahead/AlarmScreenActivity.kt

```diff
@@ -1,66 +1,138 @@
 package com.palone.planahead
 
 import android.app.KeyguardManager
-import android.content.Context
+import android.content.Intent
+import android.media.AudioAttributes
+import android.media.MediaPlayer
+import android.net.Uri
 import android.os.Build
 import android.os.Bundle
-import android.util.Log
+import android.provider.Settings
 import android.view.WindowManager
 import androidx.activity.ComponentActivity
 import androidx.activity.compose.setContent
+import androidx.annotation.RequiresApi
 import androidx.compose.foundation.layout.Arrangement
 import androidx.compose.foundation.layout.Column
 import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.material3.Button
 import androidx.compose.material3.MaterialTheme
 import androidx.compose.material3.Surface
 import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import com.palone.planahead.data.database.TaskWithAlerts
 import com.palone.planahead.data.database.alert.Alert
 import com.palone.planahead.data.database.task.Task
+import com.palone.planahead.data.database.task.properties.TaskType
+import com.palone.planahead.services.alarms.AlarmsHandler
+import dagger.hilt.android.AndroidEntryPoint
+import java.time.Duration
+import java.time.temporal.ChronoUnit
+import javax.inject.Inject
 
+@AndroidEntryPoint
 class AlarmScreenActivity : ComponentActivity() {
+    lateinit var ringtonePlayer: MediaPlayer
+
+    @Inject
+    lateinit var alarmsHandler: AlarmsHandler
+
+    @RequiresApi(Build.VERSION_CODES.P)
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         val alert = intent?.getParcelableExtra<Alert>("alert")
         val task = intent?.getParcelableExtra<Task>("task")
+
+        showScreenWhenPhoneLocked()
+        intent.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION)
+
+        val ringtoneSoundUri: Uri = Settings.System.DEFAULT_RINGTONE_URI
+        ringtonePlayer = MediaPlayer()
+
+        val audioAttributes = AudioAttributes.Builder()
+            .setUsage(AudioAttributes.USAGE_ALARM)
+            .setContentType(AudioAttributes.CONTENT_TYPE_SONIFICATION)
+            .build()
+        ringtonePlayer.setAudioAttributes(audioAttributes)
+        ringtonePlayer.isLooping = true
+        ringtonePlayer.setDataSource(this, ringtoneSoundUri)
+        ringtonePlayer.prepare()
+        ringtonePlayer.start()
```

Comment:
Can we refactor this to a separate service that is injected into `AlarmScreenActivity`? It would have two methods: `start()` and `stop()`.